### PR TITLE
feat(payments): wire up newsletter checkbox in the Checkout route

### LINKED ISSUE #9786: [subplat] Add Firefox Newsletter Opt-in Checkbox for New User Checkout Flow
'mozilla-and-you' is the newsletter.  That need to be included in the request when creating the new user.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXA-3710)

### PR DESCRIPTION
**Pull request by biancadanforth**
Because:
* When a new user is subscribing to a product, we want to provide an easy way for them to sign up for our products newsletter at the same time.

This commit:
* Building off of PR#10031 and PR#10111, POSTs to the /newsletters endpoint (for the 'mozilla-and-you' newsletter) on the auth server if an account is successfully created and the newsletter checkbox is checked.

Closes #9786

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Other information (Optional)
* I added 4 tests each for Stripe form submit and PayPal form submit in Checkout/index.test.tsx.
  * The fourth test has a TODO in it, as it is related to handling newsletter signup failure which will be covered by FXA-3667.
  * I also added "TODO: FXA-3667..." in the appropriate places to handle the error in the source code.

### CODE REVIEW

**Comment by biancadanforth** [comment]
Some storybook failures related to some prop changes I made. For some reason `yarn storybook` is building locally?

File: packages/fxa-payments-server/src/routes/Checkout/index.test.tsx [hunk 1]
```diff
@@ -347,15 +403,21 @@ describe('routes/Checkout', () => {
...
       fireEvent.change(getByTestId('new-user-confirm-email'), {
         target: { value: newAccountEmail },
       });
+      if (shouldSubscribeToNewsletter) {
+        const newsletterCheckbox = getByTestId(
+          'new-user-subscribe-product-updates'
+        );
+        fireEvent.click(newsletterCheckbox);
+        expect(newsletterCheckbox.checked).toEqual(true);
+      }
```
Comment (chenba) [inline packages/fxa-payments-server/src/routes/Checkout/index.test.tsx:420]: Do we need this?  It's the same test as the one above, and the checkbox is independent of the payment type.

File: packages/fxa-payments-server/src/routes/Checkout/index.tsx [hunk 2]
```diff
@@ -176,9 +185,23 @@ export const Checkout = ({
...
+    if (subscribeToNewsletter) {
+      try {
+        await handleNewsletterSignup();
+      } catch (error) {
+        // TODO: FXA-3667: Show AlertBar on success screen
+        // Note: If both handleNewsletterSignup and fetchProfileAndCustomer fail,
+        // there would be an AlertBar on top of the PaymentErrorView screen.
+      }
+    }
+    await fetchProfileAndCustomer();
```
Comment (chenba) [inline packages/fxa-payments-server/src/routes/Checkout/index.tsx:202]: Perhaps do this first, so the customer will see the success screen a little quicker?

**Review by chenba — APPROVED** [review]
r+ with comments

File: packages/fxa-payments-server/src/routes/Checkout/index.tsx (hunk 2 above)
Comment (biancadanforth) [reply packages/fxa-payments-server/src/routes/Checkout/index.tsx:202]: Good observation; we do call `fetchProfileAndCustomer` (via `onSuccess` callback) first for Stripe.

**Pull request merged**